compiler/rustc_lint/src/unused.rs

```diff
@@ -280,8 +280,9 @@ impl<'tcx> LateLintPass<'tcx> for UnusedResults {
                     Some(0) | None => false,
                     // If the array is definitely non-empty, we can do `#[must_use]` checking.
                     Some(n) => {
-                        let descr_pre = &format!("{}array{} of ", descr_pre, plural_suffix,);
-                        check_must_use_ty(cx, ty, expr, span, descr_pre, descr_post, n as usize + 1)
+                        let descr_pre = &format!("{}array{} of ", descr_pre, plural_suffix);
+                        let n = usize::try_from(n).unwrap_or(usize::MAX);
+                        check_must_use_ty(cx, ty, expr, span, descr_pre, descr_post, n)
```

Comment:
```
- error: unused array of arrays of arrays of `S` that must be used
+ error: unused array of array of arrays of `S` that must be used
```
maybe use `saturating_add` instead? that avoids the panic but still prints the right pluralization for any reasonable code.

Reply:
i did use `saturating` add now and also refactored everything because i didn't like it